sourcegraph/cody
# Remove auth steps

Closes #3216 

As part of the [remove authentication steps](https://docs.google.com/document/d/1AubJ226dUhaV0zC8WpA_19vzoXXtQ8AUmNqSHPaWKWQ/edit#heading=h.ftqde5tlfi9g) PRD we want to simplify the login chain a bit.

Right now, when the VS Code Cody client starts an auth processes, it will not know about the newly generated auth token until the user navigates back to VS Code. In user testing, this proofed to be confusing an some users might tab away before going through with the final redirect.

We are working on an experiment that simplifies this. The trick? When an auth process is started, start a local http server (listening on a free port on `127.0.0.1`) and forward that to the web browser. Then, when a token is created, we can `POST` the new token to that endpoint in the hope that the VS Code client.

## Test plan


https://github.com/sourcegraph/cody/assets/458591/f2ec6977-7859-4bda-be46-8d9ba2704a67

- Opt into the experiment by setting `"cody.testing.removeAuthenticationStep": true,`
- Run a local version of Cody with this PR (The accompanying dotcom change is already rollout so no other change is necessary)
- In the dev build, click one of the big sign in buttons
- Notice that you do not have to wait for the redirect-back anymore
